AssetInformationApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -27,5 +29,23 @@ public async Task<Asset> GetAssetByIdAsync(GetAssetByIdRequest query)
             var result = await _dynamoDbContext.LoadAsync<AssetDb>(query.Id).ConfigureAwait(false);
             return result?.ToDomain();
         }
+
+        [LogCall]
+        public async Task<Asset> GetAssetByAssetId(GetAssetByAssetIdRequest query)
+        {
+            _logger.LogDebug($"Calling IDynamoDBContext.QueryAsync for AssetId {query.AssetId}");
+
+            var config = new DynamoDBOperationConfig
+            {
+                IndexName = "AssetId"
+            };
+
+            var search = _dynamoDbContext.QueryAsync<AssetDb>(query.AssetId, config);
+
+            var response = await search.GetNextSetAsync().ConfigureAwait(false);
+            if (response.Count == 0) return null;
+
+            return response.First().ToDomain();
```

Comment:
What should happen if there is more than 1 result?

Reply:
There should only be 1 result, since I am querying a unique assetId.

I did try to implement it with LoadAsync, but that didnt work with a GSI.